plan_delegator/src/plan_delegator.cpp

```diff
@@ -20,6 +20,42 @@
 
 namespace plan_delegator
 {
+
+    namespace 
+    {
+        /**
+         * \brief Anonymous function to set the lanelet ids for all maneuver types except lane following. These
+         * maneuver parameters cannot be set with SET_MANEUVER_PROPERTY calls since they are not included in
+         * LANE_FOLLOW maneuvers.
+         */ 
+        void setManeuverLaneletIds(cav_msgs::Maneuver& mvr, lanelet::Id start_id, lanelet::Id end_id) {
+
+            switch(mvr.type) {
```

Comment:
We can replace this loop with a macro
 https://github.com/usdot-fhwa-stol/carma-platform/blob/23a79861f1e6b22f02a98b574f5f745db38e73a2/route_following_plugin/include/route_following_plugin.h#L48

Reply:
This macro only works for properties that are shared by all maneuver types. Since only lane_following maneuvers have the `lane_ids` parameter, and all other maneuver types have `starting_lane_id` and `ending_lane_id` parameters, these parameters require separate get/set methods. A similar function is included in route_following_plugin [here.](https://github.com/usdot-fhwa-stol/carma-platform/blob/23a79861f1e6b22f02a98b574f5f745db38e73a2/route_following_plugin/src/route_following_plugin.cpp#L55)